sys/event/event.c

```diff
@@ -96,14 +96,18 @@ event_t *event_get(event_queue_t *queue)
     return result;
 }
 
-event_t *event_wait(event_queue_t *queue)
+event_t *event_wait_multi(event_queue_t *queues, size_t n_queues)
 {
-    assert(queue);
+    assert(queues && n_queues);
     event_t *result;
 
     do {
         unsigned state = irq_disable();
-        result = (event_t *)clist_lpop(&queue->event_list);
+        for (size_t i = 0; i < n_queues; i++) {
+            result = container_of(clist_lpop(&queues[i].event_list),
```

Comment:
not sure the container_of improves the code. style I guess.

Reply:
I think `container_of()` provides some degree of type safety. (It will at least fail `list_node` is not member of `event_t`).